hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/batch/CompositeBatchPersister.java

```diff
@@ -0,0 +1,95 @@
+package com.hedera.mirror.importer.parser.batch;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Named;
+import javax.persistence.Entity;
+import javax.sql.DataSource;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.data.util.AnnotatedTypeScanner;
+
+import com.hedera.mirror.importer.domain.Upsertable;
+import com.hedera.mirror.importer.parser.CommonParserProperties;
+import com.hedera.mirror.importer.repository.upsert.UpsertQueryGenerator;
+
+@Named
+@Primary
+public class CompositeBatchPersister implements BatchPersister {
+
+    private final Map<Class<?>, BatchPersister> batchInserters = new HashMap<>();
+
+    public CompositeBatchPersister(DataSource dataSource,
+                                   MeterRegistry meterRegistry,
+                                   CommonParserProperties properties,
+                                   Collection<UpsertQueryGenerator> upsertQueryGenerators) {
+        AnnotatedTypeScanner annotatedTypeScanner = new AnnotatedTypeScanner(Entity.class);
+        Set<Class<?>> domainClasses = annotatedTypeScanner.findTypes(Upsertable.class.getPackageName());
+
+        for (Class<?> domainClass : domainClasses) {
+            Upsertable upsertable = AnnotationUtils.findAnnotation(domainClass, Upsertable.class);
+            BatchPersister batchPersister = null;
```

Comment:
nit: `BatchPersister` isn't used beyond the inserter so no need for the variable.
Just insert straight in the if/else
```suggestion
            if (upsertable != null) {
                 UpsertQueryGenerator generator = getUpsertQueryGenerator(upsertQueryGenerators, domainClass);
                 batchInserters.put(new BatchUpserter(domainClass, dataSource, meterRegistry, properties, generator));
             } else {
                 batchInserters.put(new BatchInserter(domainClass, dataSource, meterRegistry, properties));
             }
```

Reply:
I had it like that originally but that makes the `put` not DRY but also causes the line to split across multiple making the code uglier, IMO.

Reply:
Gotcha